src/main/java/com/romainplt/secretvault/restservice/SecretController.java

```diff
@@ -0,0 +1,86 @@
+package com.romainplt.secretvault.restservice;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import java.io.*;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.sql.*;
+
+@RestController
+@RequestMapping("/secretvault")
+public class SecretController {
+
+    public Secret secret;
+    public SecretService secretService;
+
+    public SecretController(SecretService secretService){
+        this.secretService = secretService;
+    }
+
+    /*
+    Print the database's hash
+     */
+    @GetMapping("/db/print")
+    public SecretPrinter printDb() throws IOException, NoSuchAlgorithmException {
+        File db = new File("/tmp/secretVault/data/database.mv.db");
+        MessageDigest shaDigest = MessageDigest.getInstance("SHA-256");
+        String shaCheckSum = HashUtils.getFileChecksum(shaDigest, db);
+        return new SecretPrinter(shaCheckSum , "database");
+    }
+
+    /*
+    Print the fspf volume's hash
+     */
+    @GetMapping("/volume/print")
+    public SecretPrinter printVolume() throws IOException, NoSuchAlgorithmException {
+        File volume = new File("/tmp/secretVault/data/volume.fspf");
+        MessageDigest shaDigest = MessageDigest.getInstance("SHA-256");
+        String shaCheckSum = HashUtils.getFileChecksum(shaDigest, volume);
+        return new SecretPrinter(shaCheckSum , "volume");
+    }
+
+    /*
+    Retrieve secrets and keys from session on CAS and :
+        - Write them in the H2 Database
+        - Print them
+     */
+    @GetMapping("/secret")
+    public SecretPrinter pushSecretFromSession(){
+        String secret=System.getenv("SECRET");
+        String key=System.getenv("KEY");
+        Secret secretSession = new Secret(secret,key);
+        secretService.writeSecretToDB(secretSession);
+        return new SecretPrinter(secretSession.getSecret() , secretSession.getKey(), "Your secret and key have " +
```

Comment:
Just remove the secretprinter and return the secret key everywhere

if(isSecretWrittent){
return Ok(secret.getKey());
} else {
return 4XX or XX
}

See response entities: baeldung.com/spring-response-entity#response-entity